### PR TITLE
[ASP-4288] Map cancelled status

### PR DESCRIPTION
**Pull request by julianaklulo**
#### What
Update the status mapping to include `cancelled` status.

#### Why
Cancelled jobs were stuck as `submitted`.

`Task`: https://jira.scania.com/browse/ASP-4288

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Review by matheushent — COMMENTED** [review]
isn't it necessary to add a database migration?

File: jobbergate-agent/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 This file keeps track of all notable changes to jobbergate-core
 
 ## Unreleased
-
+- Map job submissions with cancelled status
```
Comment (fschuch) [inline jobbergate-agent/CHANGELOG.md:6]: Please, add ticket number on the entry

Comment (julianaklulo) [reply jobbergate-agent/CHANGELOG.md]: Done!

File: jobbergate-agent/jobbergate_agent/jobbergate/constants.py [hunk 2]
```diff
@@ -19,11 +19,13 @@ class JobSubmissionStatus(str, Enum):
     SUBMITTED = "SUBMITTED"
     COMPLETED = "COMPLETED"
     FAILED = "FAILED"
+    CANCELLED = "CANCELLED"
```
[the remark this answers is not in the document]
Comment (julianaklulo) [reply jobbergate-agent/jobbergate_agent/jobbergate/constants.py:22]: Migration created: https://github.com/omnivector-solutions/jobbergate/blob/b4eb09b7eec4799d75ca2a2a36899a05bb93aa3f/jobbergate-api/alembic/versions/20240104_182646--44d41f873c8d_add_cancelled_status.py

**Review by fschuch — APPROVED** [review]

**Pull request merged**